CoViewingExample/ExampleAVPlayerSource.swift

```diff
@@ -0,0 +1,234 @@
+                                                 itemTimeForDisplay: &presentationTimestamp)
+        if let buffer = pixelBuffer {
+            if let lastTime = lastPresentationTimestamp {
+                // TODO: Use this info to target our DispatchSource timestamps?
+//                let delta = presentationTimestamp - lastTime
+//                print("Frame delta was:", delta)
+//                let movieTime = CVBufferGetAttachment(buffer, kCVBufferMovieTimeKey, nil)
+//                print("Movie time was:", movieTime as Any)
+            }
+            lastPresentationTimestamp = presentationTimestamp
+
+            guard let frame = TVIVideoFrame(timestamp: presentationTimestamp,
+                                            buffer: buffer,
+                                            orientation: TVIVideoOrientation.up) else {
+                                                assertionFailure("We couldn't create a TVIVideoFrame with a valid CVPixelBuffer.")
+                                                return
+            }
+            consumer.consumeCapturedFrame(frame)
+        }
+
+        if ExampleAVPlayerSource.useDisplayLinkTimer {
+            outputTimer?.isPaused = false
+        } else if timerSource == nil {
+            startTimerSource(hostTime: CACurrentMediaTime())
+        }
+    }
+
+    func startTimerSource(hostTime: CFTimeInterval) {
+        print(#function)
+
+        let source = DispatchSource.makeTimerSource(flags: DispatchSource.TimerFlags.strict,
+                                                    queue: sampleQueue)
+        timerSource = source
+
+        source.setEventHandler(handler: {
+            if let output = self.videoOutput {
+                let currentHostTime = CACurrentMediaTime()
+                let currentItemTime = output.itemTime(forHostTime: currentHostTime)
+                self.outputFrame(itemTimestamp: currentItemTime)
+            }
+        })
+
+        // Thread safe cleanup of temporary storage, in case of cancellation.
+        source.setCancelHandler(handler: {
+        })
+
+        // Schedule a first time source for the full interval.
+        let deadline = DispatchTime.now() + ExampleAVPlayerSource.kFrameOutputInterval
+        source.schedule(deadline: deadline,
+                        repeating: ExampleAVPlayerSource.kFrameOutputInterval,
+                        leeway: ExampleAVPlayerSource.kFrameOutputLeeway)
+        source.resume()
+    }
+
+    func addDisplayTimer() {
+        let timer = CADisplayLink(target: self,
+                                  selector: #selector(ExampleAVPlayerSource.displayLinkDidFire(displayLink:)))
+        // Fire at the native v-sync cadence of our display. This is what AVPlayer is targeting anyways.
+        timer.preferredFramesPerSecond = 0
+        timer.isPaused = true
+        timer.add(to: RunLoop.current, forMode: RunLoop.Mode.common)
+        outputTimer = timer
+    }
```

Comment:
Given that AVPlayer is trying to present video on our display, and targets its natural cadence I think it is the best to use a CADisplayLink timer at this point. However, we don't necessarily need to run this timer on the main thread. We could spawn another NSThread, and use it exclusively to service DisplayLink callbacks.